Assets/_Scripts/FieldState.cs

```diff
@@ -3,101 +3,34 @@
 using UnityEngine;
 using UnityEngine.UI;
 using System.Linq;
-using UnityEngine.SceneManagement;
 
+// Class that holds field information / helper functions
 public class FieldState : MonoBehaviour
-{
-    private Enemy[] _enemies = new Enemy[Party.numPlayers];
+{   
+    public Party<Player> playerParty = new Party<Player>();
+    public Party<Enemy> enemyParty = new Party<Enemy>(); 
 
 
     [Header("Slots")]
-    public Transform heroSlot;
-    public List<Transform> mookSlots;
-    public List<Transform> enemySlots;
+    public FighterSlot heroSlot;
+    public List<FighterSlot> mookSlots;
+    public List<FighterSlot> enemySlots;
 
     public int currentWaveIndex {get; set;}
 
-    Player _heroPlayer;
-
-
     public void Initialize() {
-        Messenger.AddListener<DeathResult>(Messages.OnEntityDeath, this.onEntityDeath);
-        Messenger.AddListener(Messages.OnWaveComplete, this.onWaveComplete);
-        GameManager.Instance.party.Initialize();
 
         this.InitializePlayers();
         this.InitializeEnemies();
```

Comment:
Initialize these fields in TurnController as well.

Reply:
Disagree - see above.

Reply:
Similarly, we could add a one-time-phase `OnStageStart` in TurnController (or some subsidiary of this) which initializes the parties.

Reply:
That's kind of what TurnController.Initialize() does right now. I don't want to make initialization a message right now because it should be well defined when it happens. Also, you must register the listener before an object can listen to it, so there might be some discrepancies in ordering.